util/FINETUNE_LM_PARAMETERS.md

```diff
@@ -0,0 +1,43 @@
+# Finetune Language Model Parameters: Alpha, Beta
+This topic is to explain how did `finetune_lm_params.py` fit `alpha` and `beta`
+
+## Inspiration
+According to mesh plot result of [Paddle/DeepSpeech-Grid-Search](https://github.com/PaddlePaddle/DeepSpeech/raw/develop/docs/images/tuning_error_surface.png), I assume `BatchWER v.s. Alpha` is `ideal valley parabolic curve`, i.e. which fits this formula below:
+
+Idea Valley Parabola Formula
+===
+- Assume there exists a `lowest WER` satisfies the formula: ![A.1](https://latex.codecogs.com/gif.latex?WER_{batch}=A_0(\\alpha-\\alpha_{best})^2+A_1 "A.1"),  `(A.1)`
+- Where ![](https://latex.codecogs.com/gif.latex?A_0>0) and ![](https://latex.codecogs.com/gif.latex?A_1>0)
+- If we set ![](https://latex.codecogs.com/gif.latex?\\alpha=\\alpha_{best}), the `WER` should be lowest: ![](https://latex.codecogs.com/gif.latex?WER_{lowest}=A_1), so the target is to find out the constant ![](https://latex.codecogs.com/gif.latex?\\alpha_{best})
+- To solve the `best alpha`, I decompose the formula, and change the coefficient as below:
+- ![A.2](https://latex.codecogs.com/gif.latex?WER_{batch}=A_0\\alpha^2-2A_0\\alpha_{best}\\alpha+A_1\\alpha_{best}^2+A_1 "A.2"),  `(A.2)`
+
+Polynomial Idea Valley Parabola Formula
+===
+- Let's tidy the `(A.2)` coefficients as below:
+- Assume there exists a `lowest WER` satisfies the formula: ![B.1](https://latex.codecogs.com/gif.latex?WER_{Batch}=B_0\\alpha^2+B_1\\alpha+B_2 "B.1"),  `(B.1)`
+- Where ![](https://latex.codecogs.com/gif.latex?B_0>0,B_1<0,B_2>0)
+- Compare to `(A.1)`: ![](https://latex.codecogs.com/gif.latex?B_0=A_0) and ![](https://latex.codecogs.com/gif.latex?B_1=-2A_0\\alpha_{best})
```

Comment:
@mychiux413 You don't really want to say here that B_0=A_0. You want to say that B_0 has an analogous role to A_0 in the alpha quadratic equation.

Similarly you don't want to say B_1 = -2A_0\\alpha_{best}. You want to say that B_1 has an analogous role to -2A_0\\alpha_{best} in the alpha quadratic equation.